### PR TITLE
fix: pkg/errors is no longer maintained

### LINKED ISSUE #7402: pkg/errors is no longer maintained
# Summary

pkg/errors is no longer maintained. See https://github.com/pkg/errors/issues/245

# Use Cases

We can replaces all calls to errors.Wrap/f with fmt.Errorf and removes the unneeded dependency on
github.com/pkg/errors since we don't support go 1.12 or lower.  See https://github.com/microsoft/go-winio/pull/227 for an example.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by stephpalis**
Signed-off-by: Stephanie Palis <stephanie.palis@acquia.com>
Fixes #7402

Don't bother creating a PR until you've done this:

* [x] Run `make pre-commit -B` to fix codegen, lint, and commit message problems.

Create your PR as a draft.

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it
  does not need to pass.
* Once required tests have passed, you can make it "Ready for review".
* Say how how you tested your changes. If you changed the UI, attach screenshots.

Tips:

* If changes were requested, and you've made them, then dismiss the review to get it looked at again.
* Add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md) if you like.
* You can ask for help!
___________________________________________________________________________________
This PR removes the functionality of StackTracer as a result of removing pkg/errors

### CODE REVIEW

File: errors/errors_test.go [hunk 1]
```diff
@@ -25,30 +19,28 @@ func TestErrorf(t *testing.T) {
...
-	assert.Equal(t, "hello world", intWrap.Error())
+	assert.Equal(t, "wrapped message: original error message", argoErr.Error())
 }
 
-func TestStackTrace(t *testing.T) {
-	err := errors.New("MYCODE", "my message")
-	assert.Contains(t, fmt.Sprintf("%+v", err), "errors_test.go")
-}
+//// TestInternalError verifies
+//func TestInternalError(t *testing.T) {
```
Comment (alexec) [inline errors/errors_test.go:26]: clean-up?

File: util/errors/errors.go [hunk 2]
```diff
@@ -12,15 +12,12 @@ import (
...
 	apierr "k8s.io/apimachinery/pkg/api/errors"
-
-	argoerrs "github.com/argoproj/argo-workflows/v3/errors"
 )
 
 func IsTransientErr(err error) bool {
 	if err == nil {
 		return false
 	}
-	err = argoerrs.Cause(err)
```
Comment (alexec) [inline util/errors/errors.go:23]: I think we need to be careful here, we should probably be unwrapping errors. Funcs like `isExcheedeQuotaErr` maybe need to be updated to use `errors.Is()`

File: util/kubeconfig/kubeconfig.go [hunk 3]
```diff
@@ -43,15 +44,15 @@ func GetRestConfig(token string) (*restclient.Config, error) {
 		token = strings.TrimSpace(strings.TrimPrefix(token, BasicAuthScheme))
 		username, password, ok := decodeBasicAuthToken(token)
 		if !ok {
-			return nil, errors.New("Error parsing Basic Authentication")
+			return nil, errors.New("", "Error parsing Basic Authentication")
```
Comment (alexec) [inline util/kubeconfig/kubeconfig.go:47]: I think the first argument should be something?

File: workflow/executor/resource.go [hunk 4]
```diff
@@ -216,7 +216,6 @@ func (we *WorkflowExecutor) checkResourceState(ctx context.Context, selfLink str
 	stream, err := request.Stream(ctx)
 
 	if err != nil {
-		err = errors.Cause(err)
```
Comment (alexec) [inline workflow/executor/resource.go:219]: this line is purposeful, need to fine alternative?

File: errors/errors.go [hunk 5]
```diff
@@ -26,28 +23,20 @@ type ArgoError interface {
...
-	StackTrace() errors.StackTrace
+	err     error
 }
 
 // New returns an error with the supplied message.
 // New also records the stack trace at the point it was called.
 func New(code string, message string) error {
-	err := errors.New(message)
-	return argoerr{code, message, err.(stackTracer)}
+	err := fmt.Errorf(message)
```
Comment (alexec) [inline errors/errors.go:38]: should we moving to `Errorf("%w")` and `errors.Is()`

**Review by alexec — CHANGES_REQUESTED** [review]

File: errors/errors.go (hunk 5 above)
Comment (stephpalis) [reply errors/errors.go]: @alexec `Errorf("%w")` is used in the `Wrap` function. I didn't find a need to use `errors.Is()` and instead relied on Go's error library

File: util/kubeconfig/kubeconfig.go [hunk 6]
```diff
@@ -2,6 +2,7 @@ package kubeconfig
 
 import (
 	"encoding/base64"
+	goErrors "errors"
```
Comment (blkperl) [inline util/kubeconfig/kubeconfig.go:5]: Is there a conflict or did your editor automatically do this before the errors package was removed?

**Review by alexec — COMMENTED** [review]
In case this changes behaviour, lets hold off merging until v3.4

**Comment by blkperl** [comment]
@alexec 3.3 is tagged, are we good to merge this?

**Comment by alexec** [comment]
needs conflict resolution

**Comment by stephpalis** [comment]
@alexec pushed up a commit, if you could retrigger the test suite

**Review by alexec — APPROVED** [review]

**Pull request merged**